src/cli/telemetry/telemetry.ts

```diff
@@ -144,21 +102,55 @@ export async function telemetryAction(action?: TelemetryCallback) {
   }
 }
 
-function hasAppTarget(): boolean {
+export function hasAppTarget(): boolean {
   return getStencilCLIConfig().validatedConfig.config.outputTargets.some(
-    target => target.type === 'www' && (!!target.serviceWorker || target.baseUrl !== '/'),
+    target => target.type === 'www' && (!!target.serviceWorker || (!!target.baseUrl && target.baseUrl !== '/')),
   );
 }
 
-function isUsingYarn() {
+export function isUsingYarn() {
   return getCompilerSystem().getEnvironmentVar('npm_execpath')?.includes('yarn') || false;
 }
 
-async function getActiveTargets(): Promise<string[]> {
+export async function getActiveTargets(): Promise<string[]> {
   const result = getStencilCLIConfig().validatedConfig.config.outputTargets.map(t => t.type);
   return Array.from(new Set(result));
 }
 
+export const prepareData = async (duration_ms: number, component_count: number = undefined): Promise<TrackableData> => {
+  const { flags, sys } = getStencilCLIConfig();
+  const { typescript, rollup } = getCoreCompiler()?.versions || { typescript: 'unknown', rollup: 'unknown' };
+  const packages = await getInstalledPackages() || [];
```

Comment:
I _think_ this always returns an array, and don't think we need the `|| []` here